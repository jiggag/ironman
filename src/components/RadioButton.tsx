import React, { memo } from 'react';
import { Text, TouchableOpacity, StyleSheet } from 'react-native';
import { View } from 'react-native-ui-lib';
import Constant from '@utils/constants';

interface RadioButtonProps {
  onPress: () => void;
  value: string;
  isSelected: boolean;
}

export const RadioButton = memo<RadioButtonProps>(({ onPress, value, isSelected }) => (
  <TouchableOpacity activeOpacity={0.5} style={styles.button} onPress={onPress}>
    <View row>
      <View style={[styles.radio, isSelected && styles.selectedRadio]} />
      <Text style={styles.valueText}>{value}</Text>
    </View>
  </TouchableOpacity>
));

const styles = StyleSheet.create({
  button: {
    alignItems: 'center',
  },
  radio: {
    backgroundColor: Constant.WHITE_COLOR,
    borderRadius: 14,
    borderWidth: 1,
    elevation: 5,
    height: 14,
    shadowColor: Constant.SHADOW_COLOR,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.2,
    shadowRadius: 2,
    width: 14,
  },
  selectedRadio: {
    backgroundColor: Constant.MAIN_COLOR,
  },
  valueText: {
    color: Constant.BLACK,
    fontSize: 12,
    marginLeft: 2,
    marginRight: 8,
  },
});
