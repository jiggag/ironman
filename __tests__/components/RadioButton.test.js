import React from 'react';
import { render } from '@testing-library/react-native';
import RadioButton from '../../src/components/RadioButton';

it('Renderer Component: RadioButton', () => {
  const { baseElement } = render(
    <RadioButton onPress={jest.fn()} value="라디오버튼" isSelected />
  )
  expect(baseElement).toMatchSnapshot();
});