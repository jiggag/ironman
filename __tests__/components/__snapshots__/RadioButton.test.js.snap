// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderer Component: RadioButton 1`] = `
<View
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "opacity": 1,
    }
  }
>
  <View
    row={true}
    style={
      Array [
        undefined,
        undefined,
        undefined,
        Object {},
        Object {},
        Object {
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          undefined,
          undefined,
          undefined,
          Object {},
          Object {},
          Object {},
          Array [
            Object {
              "backgroundColor": "#ffffff",
              "borderRadius": 14,
              "borderWidth": 1,
              "elevation": 5,
              "height": 14,
              "shadowColor": "#333333",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 2,
              "width": 14,
            },
            Object {
              "backgroundColor": "#ec2",
            },
          ],
        ]
      }
    />
    <Text
      style={
        Object {
          "fontSize": 12,
          "marginLeft": 2,
          "marginRight": 8,
        }
      }
    >
      라디오버튼
    </Text>
  </View>
</View>
`;
